pkg/upgrade/upgrades/backfill_jobs_info_table_migration.go

```diff
@@ -12,37 +12,70 @@ package upgrades
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/cockroachdb/cockroach/pkg/clusterversion"
 	"github.com/cockroachdb/cockroach/pkg/jobs"
 	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
 	"github.com/cockroachdb/cockroach/pkg/upgrade"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/errors"
 )
 
-const backfillJobInfoTableWithPayloadStmt = `
-INSERT INTO system.job_info (job_id, info_key, value)
-SELECT id, $1, payload FROM system.jobs
-`
+const (
+	backfillJobInfoSharedPrefix = `WITH inserted AS (
+		INSERT INTO system.job_info (job_id, info_key, value) 
+	SELECT id, '`
 
-const backfillJobInfoTableWithProgressStmt = `
-INSERT INTO system.job_info (job_id, info_key, value)
-SELECT id, $1, progress FROM system.jobs
-`
+	backfillJobInfoSharedSuffix = ` FROM system.jobs 
+	WHERE jobs.id > $1
+	ORDER BY jobs.id ASC
+	LIMIT $2
+	RETURNING job_id) SELECT job_id FROM inserted ORDER BY job_id DESC LIMIT 1`
+
+	backfillJobInfoPayloadStmt  = backfillJobInfoSharedPrefix + jobs.LegacyPayloadKey + `', payload` + backfillJobInfoSharedSuffix
+	backfillJobInfoProgressStmt = backfillJobInfoSharedPrefix + jobs.LegacyProgressKey + `', progress` + backfillJobInfoSharedSuffix
+)
 
 func backfillJobInfoTable(
 	ctx context.Context, cs clusterversion.ClusterVersion, d upgrade.TenantDeps,
 ) error {
-	return d.DB.KV().Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
-		_, err := d.InternalExecutor.ExecEx(ctx, "backfill-job-info-payload", txn,
-			sessiondata.NodeUserSessionDataOverride, backfillJobInfoTableWithPayloadStmt, jobs.GetLegacyPayloadKey())
-		if err != nil {
-			return errors.Wrap(err, "failed to backfill legacy payload")
-		}
+	const batchSize = 100
```

Comment:
Should we make this a non-public cluster setting incase we need fiddle with it in the future?

Reply:
ugh, a registered  *cluster setting* feels like total overkill for a one-time job. what kind of fiddling you thinking? lowering to like, 10 if the table is under heavy writes? I figured 100 should be plenty small enough to get it in in-between writes.

I could _maybe_ see env var but cluster setting feels heavy to me?

Reply:
yeah i just don't want to end up wishing we had some way of tuning this down to 1 or 10 and not being able to. env var is totally fine with me. It's then sufficient to run the test I mentioned with that env var set to anywhere from 1 to 5 to test the resume after logic.

Reply:
env var now